File: old/src/stack/stack_menu.cpp
#include <iostream>

#include "stack_menu.hpp"
#include "stack.hpp"

using namespace std;

void StackMenu::show() {
    Stack stack1;
    StackItem item;
    int option;

    cout << "\nStack generator\n";

    do
    {
        cout << "1 - Insert element.\n";
        cout << "2 - Remove element.\n";
        cout << "3 - Print stack.\n";
        cout << "4 - Print stack length.\n";
        cout << "\n0 - Stop program.\n";

        cin >> option;

        if (option == 1)
        {
            cout << "Write element to insert: \n";
            cin >> item;

            stack1.push(item);
        }
        else if (option == 2)
        {
            item = stack1.pop();

            cout << "Removed element: " << item << endl;
        }
        else if (option == 3)
        {
            stack1.print();
        }
        else if (option == 4)
        {
            stack1.stackSize();
        } else if (option == 0) {
            exit(0);
        } else {
            cout << "Invalid option!\n";
        }
    } while (option != 0);
}

File: src/stack_mod/stack_menu.rs
use std::io;

use super::stack::Stack;

pub fn start_stack() {
    let mut stack: Stack = Stack::new();

    fn load_argument() -> i32 {
        let mut option: String = String::new();

        io::stdin()
            .read_line(&mut option)
            .expect("Failed to read line");
        option.trim().parse().expect("Please type a number!")
    }

    println!("Stack generator");

    loop {
        println!("1 - Insert element");
        println!("2 - Remove element");
        println!("3 - Print stack");
        println!("4 - Print stack length");
        println!("0 - Stop program");

        let option: i32 = load_argument();

        match option {
            1 => {
                println!("Write element to insert: ");
                let element: i32 = load_argument();
                stack.push(element);
                continue;
            }
            2 => {
                let removed_element: i32 = stack.pop();
                println!("Removed element: {}", removed_element);
                continue;
            }
            3 => {
                println!("Print stack");
                stack.print();
            }
            4 => {
                println!("Print stack length");
                let stack_size: i32 = stack.get_stack_size();
                println!("Stack size: {}", stack_size);
                continue;
            }
            0 => {
                println!("Program stopped!");
                return;
            }
            _ => {
                println!("Invalid option!");
                continue;
            }
        }
    }
}
